Config cleanup, specify host and port

guggero opened the pull request:
This PR proposes the following changes:
* Format with `go fmt`.
* Remove everything from `config.go` that is not actually used to parse either command line options and/or LND's configuration file.
* Hide irrelevant options/groups from help output.
* Add new option `--host` (or short `-h`) to specify a specific host.
* Add new option `--port` (or short `-p`) to specify the port to use (default: `10009`).

The only possibly breaking change is that the port is not read from the `--rpclisten` option any more but needs to be specified explicitly. In my opinion it was not really transparent that the port was taken from this option in the first place. So having an explicit parameter would be much more clear to me.
But I can revert this if wished.

guggero committed:
Config cleanup, specify host and port

ottosuess commented:
awesome, thank you.

the reason i kept all the unrelated config stuff in `config.go` is that i got weird results from `go-flags` when it encountered unknown items and groups in the config file.

I think it stops parsing the config file on the first unknown item or something like that.

Did you run into this problem?

guggero commented:
Yes, I noticed that as soon as I removed the extra stuff.
But using the custom parser with the ignore unknown option helped:
```
	p := flags.NewParser(&cfg, flags.IgnoreUnknown)
	if err := flags.NewIniParser(p).ParseFile(cfg.ConfigFile); err != nil {
```
Then it only complained about the groups that it wasn't able to find in the struct.
So I added them but hidden. That way they don't appear in the help screen:
```
	// The following lines we only need to be able to parse the
	// configuration INI file without errors. The content will be ignored.
	BtcdMode      *chainConfig `hidden:"true" group:"btcd" namespace:"btcd"`
	BitcoindMode  *chainConfig `hidden:"true" group:"bitcoind" namespace:"bitcoind"`
	NeutrinoMode  *chainConfig `hidden:"true" group:"neutrino" namespace:"neutrino"`
	LtcdMode      *chainConfig `hidden:"true" group:"ltcd" namespace:"ltcd"`
	LitecoindMode *chainConfig `hidden:"true" group:"litecoind" namespace:"litecoind"`
	Autopilot     *chainConfig `hidden:"true" group:"Autopilot" namespace:"autopilot"`
	Tor           *chainConfig `hidden:"true" group:"Tor" namespace:"tor"`
	Hodl          *chainConfig `hidden:"true" group:"hodl" namespace:"hodl"`
```

It's not very elegant but seems to work pretty well.

ottosuess commented:
got it.

this is really cool. thanks for making the pr! 😀
